Lib/datetime.py

```diff
@@ -1572,6 +1573,14 @@ def _fromtimestamp(cls, t, utc, tz):
             # 23 hours at 1969-09-30 13:00:00 in Kwajalein.
             # Let's probe 24 hours in the past to detect a transition:
             max_fold_seconds = 24 * 3600
+
+            # On Windows localtime_s throws an OSError for negative values,
+            # thus we can't perform fold detection for values of time less
+            # than the max time fold. See comments in _datetimemodule's
+            # version of this method for more details.
+            if sys.platform.startswith("win") and t < max_fold_seconds:
```

Comment:
Can we take the call to `sys.platform.startswith()` outside the function?  Or at least change the order of `and` operands above so that it is not  called every time?

Reply:
Changed the order seeing as `t < max_fold_seconds` is a pretty rarely going to be true.